src/server/api/redirect.ts

```diff
@@ -171,18 +180,52 @@ export default async function redirect(req: Express.Request, res: Express.Respon
     if (gaTrackingId) gaLogging(req, res, shortUrl, longUrl)
 
     // Redirect immediately if a crawler is visiting the site
-    if (isCrawler(req.headers["user-agent"] || '')) {
+    if (isCrawler(req.headers['user-agent'] || '')) {
       res.status(302).redirect(longUrl)
       return
-    } else {
-      // Serve transition page
-      // TODO: Get/set a browser cookie so that this isn't served too frequently
+    }
+
+    // Reassure typescript that the session object exists.
+    if (!req.session) {
+      throw new Error('Session object does not exist')
+    }
+
+    if (!req.session.visits || !req.session.visits[shortUrl]) {
+      // This is the first time visiting a/the shortlink.
+      req.session.visits = {
+        ...req.session.visits,
+        [shortUrl]: Math.floor(Date.now()),
+      }
+
+      // Avoid exceeding cookie size limit by pruning
+      // old entries.
```

Comment:
I have a concern with limiting by an arbitrary number of short links, which is susceptible to variability in the length of each short link, when we are able to compute exactly the number of bytes that will be taken up.

As the characters that make up the short links and epoch are a subset of ASCII, the size in bytes is also the length of the string. Can we use that instead to guarantee keeping under the 4k limit?

We might also leave some buffer and use just e.g. 3k.

FInally, could this be extracted into a separate function?